lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -275,6 +275,21 @@ def get_cpu_facts(self, collected_facts=None):
                 cpu_facts['processor_vcpus'] = (cpu_facts['processor_threads_per_core'] *
                                                 cpu_facts['processor_count'] * cpu_facts['processor_cores'])
 
+                if sys.version_info[0] >= 3:
+                    cpu_facts['processor_nproc'] = len(
+                        os.sched_getaffinity(0)
+                    )
+                else:
+                    cmd = ('nproc',)
+                    rc, out, _ = self.module.run_command(cmd)
```

Comment:
This needs to go through `get_bin_path()` in order to get the absolute path to the executable since the execution `PATH` in the remote environment is far from consistent.
```suggestion
                   try:
                       cmd = get_bin_path('nproc')
                   except ValueError:
                       cpu_facts['processor_nproc'] = processor_occurence
                       
                    rc, out, err = self.module.run_command(cmd)
```

Reply:
very good, thanks